novawallet/Modules/NetworkDetails/NetworkDetailsInteractor.swift

```diff
@@ -166,6 +197,93 @@ private extension NetworkDetailsInteractor {
             nodesConnections[node.url] = connection
         }
     }
+
+    func disconnectNodes() {
+        nodesConnections.values.forEach { connection in
+            connection.disconnect(true)
+        }
+
+        nodesConnections = [:]
+    }
+
+    func measureNodePing(for nodeUrl: URL) {
+        guard
+            let connection = nodesConnections[nodeUrl.absoluteString],
+            let key = try? StorageKeyFactory().accountInfoKeyForId(
+                AccountId.zeroAccountId(of: chain.accountIdSize)
+            )
+        else {
+            return
+        }
+
+        let measureOperation: BaseOperation<Int> = if chain.isEthereumBased {
+            createPingMeasureOperation(
+                with: createEVMQueryOperation(with: connection),
+                queue: operationQueue
+            )
+        } else {
+            createPingMeasureOperation(
+                with: storageRequestFactory.queryOperation(
+                    for: { [key] },
+                    at: nil,
+                    engine: connection
+                ),
+                queue: operationQueue
+            )
+        }
+
+        execute(
+            operation: measureOperation,
+            inOperationQueue: operationQueue,
+            runningCallbackIn: .main
+        ) { [weak self] result in
+
+            switch result {
+            case let .success(ping):
+                self?.presenter?.didReceive(
+                    .pinged(ping),
+                    for: nodeUrl.absoluteString
+                )
+            case let .failure(error):
+                print(error)
```

Comment:
Let's communicate the error to the presenter. Probably, in that can we can switch the node to the connecting state or discuss with the designer a new state